src/System.Net.Security/src/System/Net/NegotiateStreamPal.Unix.cs

```diff
@@ -134,7 +213,32 @@ internal static Exception CreateExceptionFromError(SecurityStatusPal statusCode)
             out int newOffset,
             uint sequenceNumber)
         {
-            throw new PlatformNotSupportedException();
+            if (offset < 0 || offset > (buffer == null ? 0 : buffer.Length))
```

Comment:
So long as count is zero, sure.  Really it's for the edge case where Length is 0.  Not sure it applies in this specific case, but it's the normal offset/count/length bounds check.